spec.html

```diff
@@ -21,20 +21,24 @@ <h1>Object.fromEntries ( iterable )</h1>
     1. Let _obj_ be ObjectCreate(%ObjectPrototype%).
     1. Assert: _obj_ is an extensible ordinary object with no own properties.
     1. Let _iter_ be ? GetIterator(_iterable_).
-    1. Repeat,
-      1. Let _next_ be ? IteratorStep(_iter_).
-      1. If _next_ is *false*, return _obj_.
-      1. Let _nextItem_ be ? IteratorValue(_next_).
-      1. If Type(_nextItem_) is not Object, then return ? IteratorClose(_iter_, ThrowCompletion(a newly created *TypeError* object)).
-      1. Let _key_ be Get(_nextItem_, `"0"`).
-      1. If _key_ is an abrupt completion, return ? IteratorClose(_iter_, _key_).
-      1. Let _key_ be _key_.[[Value]]
-      1. Let _value_ be Get(_nextItem_, `"1"`).
-      1. If _value_ is an abrupt completion, return ? IteratorClose(_iter_, _value_).
-      1. Let _value_ be _value_.[[Value]]
-      1. Let _propertyKey_ be ToPropertyKey(_key_).
-      1. If _propertyKey_ is an abrupt completion, return ? IteratorClose(_iter_, _propertyKey_).
-      1. Let _propertyKey_ be _propertyKey_.[[Value]]
-      1. Peform ! CreateDataPropertyOrThrow(_obj_, _propertyKey_, _value_).
+    1. If _iter_ is *undefined* or *null*, return _obj_.
+    1. Let _stepsDefine_ be the algorithm steps defined in <emu-xref href="#sec-create-data-property-functions">CreateDataProperty Functions</emu-xref>.
+    1. Let _adder_ be CreateBuiltinFunction(_stepsDefine_, &laquo; &raquo;).
+    1. Return ? AddEntriesFromIterable(_obj_, _iterable_, _adder_).
   </emu-alg>
+  <emu-note>
+    The function created for _adder_ is never directly accessible to ECMAScript code.
+  </emu-note>
+</emu-clause>
+
+<emu-clause id="sec-create-data-property-functions">
+  <h1>CreateDataProperty Functions</h1>
+  <p>A CreateDataProperty function is an anonymous built-in function. When a CreateDataProperty function is called with arguments _key_ and _value_, the following steps are taken:</p>
```

Comment:
This sounds like we're defining a new kind of function. Maybe word it to say that we're talking about a specific one?

Additionally, this name is problematic as there's already a [`CreateDataProperty` abstract operation](https://tc39.github.io/ecma262/#sec-createdataproperty).

Reply:
Ah, I'll rename it - the wording here mirrors that of CreateThenFinally/CreateCatchFinally.